services/QuillLMS/client/app/bundles/Grammar/actions/session.ts

```diff
@@ -127,10 +127,16 @@ const normalizeQuestionWithAttempts = (question: object): object => {
   }
 }
 
+const getUID = () => {
+  const url = window.location.href;
+  const uidMatch = url.match(/[?&]uid=([^&]+)/);
+  return uidMatch ? uidMatch[1] : null;
+}
+
```

Comment:
There's a `getParameterByName` function you can use from the helpers (`client/app/bundles/Grammar/helpers/getParameterByName.ts`)

Reply:
Eric is right -- example of how this being used here: https://github.com/empirical-org/Empirical-Core/blob/bfb249482a998f2e29a51c6343f76c9b0745f159/services/QuillLMS/client/app/bundles/Grammar/components/grammarActivities/container.tsx#L90.